packages/backend/services/propertyService.js

```diff
@@ -1,61 +1,82 @@
+        const propertyCoordinatesWithinPolygon = turf.pointsWithinPolygon(points, searchWithin);
+        //turf give us back an array of object, we go through it and we find amongst all the properties, the ones that have the same coordinates, and we add it to the array
+        if (propertyCoordinatesWithinPolygon.features.length) {
+            propertyCoordinatesWithinPolygon.features.forEach((feature) => {
+                let property = getAllProperties().find(
+                    (property) => property.details.coordinates === feature.geometry.coordinates,
+                );
+                if (property) {
+                    propertiesWithinPolygons.push({
+                        ...property,
+                        leafletId: id,
+                    });
+                }
+            });
         }
-      });
-    }
-  });
-  return [...new Set(propertiesWithinPolygons.flat())];
+    });
+    propertiesWithinPolygons.forEach((property) => {
+        calculateTravelToWork(property);
+    });
+    return [...new Set(propertiesWithinPolygons.flat())];
 };
 
-const getDistance = async (startCoordinates, endCoordinates, methodOfTravel) => {
-  const token = `?access_token=${process.env.MAPBOX_TOKEN}`;
-
-  const { data } = await axios.get(
-    `https://api.mapbox.com/directions/v5/mapbox/${methodOfTravel}/${startCoordinates}${endCoordinates}?access_token=${token}`
-  );
+const getDirections = async (startCoordinates, endCoordinates, methodOfTravel) => {
+    const token = process.env.MAPBOX_TOKEN;
+    const { data } = await axios.get(
+        `https://api.mapbox.com/directions/v5/mapbox/${methodOfTravel}/${startCoordinates};${endCoordinates}?access_token=${token}`,
+    );
+    return data.routes;
+};
 
-  data.routes.forEach((route) => {
-    route.durationByTrain = calculateTrainDuration(route.distance, route.duration);
-    route.durationByBus = calculateBusDuration(route.distance, route.duration);
-  });
-  return data;
+const getNearestPointOfInterest = async (pointOfInterest, coordinates) => {
+    const resp = await axios.get(
+        `https://api.mapbox.com/geocoding/v5/mapbox.places/${pointOfInterest}.json?type=poi&proximity=${coordinates}&access_token=${token}`,
+    );
+    const data = [
+        resp.data.features[0].properties.address,
+        resp.data.features[0].context[0].text,
+        resp.data.features[0].geometry.coordinates,
+    ];
+    return data;
 };
 
-//TOTO what does this???
-const getNearestWhat = async (pointOfInterest, coordinates) => {
-  const url = `https://api.mapbox.com/geocoding/v5/mapbox.places/${pointOfInterest}.json?type=poi&proximity=${coordinates}&access_token=${token}`;
-  const resp = await axios.get(url);
-  return resp.data;
+const calculateTravelToWork = async (property) => {
+    const endCoordinates = AND_DIGITAL_COORDINATES;
+
+    const startCoordinates = [property.details.coordinates[1], property.details.coordinates[0]];
+
+    for (const i of METHOD_OF_TRAVEL) {
```

Comment:
pickiness... const methodOfTravel of METHODS_OF_TRAVEL would be nicer. up to you, though